### ISSUE TITLE
Batch 1: Publish Find VA Form detail pages

### ISSUE THREAD

**Issue by MickinSahni** [department-of-veterans-affairs/va.gov-team#11155]
## Issue Description
Given we need to publish the Find VA Form detail pages via Drupal
we want to publish them in batches
so that we safely launch the detail pages. 

---
## Tasks
- [x] Define the batches and the forms that'll be contained within 
- [x] Add basic-auth gate
-  [x] Publish the pages via Drupal

## Acceptance Criteria
- Form detail pages are safely launched
 - Publish batch 1: Top 30-40 identified MVP form pages.

## 

Batch 1 form pages:

1.	/find-forms/about-form-21-4138 
2.	/find-forms/about-form-22-1995  
3.	/find-forms/about-form-21-526ez 
4.	/find-forms/about-form-21-2680 
5.	/find-forms/about-form-10-7959c  
6.	/find-forms/about-form-10-10d  
7.	/find-forms/about-form-21-22  
8.	/find-forms/about-form-10-5345  
9.	/find-forms/about-form-20-0995  
10.	/find-forms/about-form-21-686c  <-- Online tool hasn’t launched
11.	/find-forms/about-form-10-0137 
12.	/find-forms/about-form-21-0781  
13.	/find-forms/about-form-21-0966   
14.	find-forms/about-form-21-0845  
15.	/find-forms/about-form-22-5490 
16.	/find-forms/about-form-26-1880  
17.	/find-forms/about-form-10-2850c  
18.	/find-forms/about-form-21-4142 
19.	/find-forms/about-form-21p-0969  
20.	/find-forms/about-form-21p-534ez  
21.	/find-forms/about-form-10-10ez-pdf  
22.	/find-forms/about-form-10-3542  
23.	/find-forms/about-form-22-1990e  
24.	/find-forms/about-form-22-1990n  
25.	/find-forms/about-form-22-1990  
26.	/find-forms/about-form-22-5495  
27.	/find-forms/about-form-va40-10007  
28.	/find-forms/about-form-21p-530  
29.	find-forms/about-form-21p-527ez  
30.	/find-forms/about-form-10-10cg  <-- Online tool hasn’t launched
31.	/find-forms/about-form-21-22a  
32.	/find-forms/about-form-21-674  
33.	/find-forms/about-form-21p-509  
34.	/find-forms/about-form-10-0137-espanol  
35.	/find-forms/about-form-10-0137a  
36.	/find-forms/about-form-21-0781a  
37.	/find-forms/about-form-21p-527 
38.	/find-forms/about-form-21p-534  
39.	/find-forms/about-form-10-10ez (esp)  
40.	/find-forms/about-form-10-10ezr  
41.	find-forms/about-form-21-0779  

**Dependency:**
Because of transformer and processing memory dependency, we will have the pages ready in Draft CMS state and check memory/transformer situation before publishing. (We may need to publish batches as separate tickets.)

**Note:** may be able to soft launch without having redirects completed first to at least verify results to detail page transition. In that case, we can remove the blocker on this ticket (see Detail Page Release Plan ticket for further details #11311)

**Comment by MickinSahni** [comment]
i created this ticket but see that it might duplicate #10819.
can someone confirm whether there's a need for a separate publishing ticket (this one) or will #10819 suffice?

**Comment by jenniferlee-dsva** [comment]
#10819 includes batch publishing as AC, but we can separate them out - they are sort of 2 separate tasks, and the publishing step has some dependencies that creating the FE template doesn't have. I can strike out the publish AC on #10819 @MickinSahni

**Comment by jenniferlee-dsva** [comment]
Moved the batch publishing AC from #10819 to this ticket.

**Comment by jenniferlee-dsva** [comment]
fysa @bethpottsVADEPO - If basic auth ticket is completed while I'm OOO, the above 41 forms can be set to publish by Selina, if she has bandwidth. (Otherwise I will batch publish these when I return 9/9 - which is fine, too - just don't want to be the source of holding things up.)

**Comment by ncksllvn** [comment]
Basic auth is ready, https://github.com/department-of-veterans-affairs/va.gov-team/issues/13125, so this issue is unblocked

**Comment by lunascoop** [comment]
I set Form 21-4138 (#1 in the batch list above) to publish in Drupal as a trial run.

**Comment by ncksllvn** [comment]
https://www.va.gov/find-forms/about-form-21-4138/ has been published! Use the prod credentials to access it, and make sure you include the trailing slash.

**Comment by lunascoop** [comment]
The remaining pages have been set to publish. Will validate after today's deployment.

**Comment by lunascoop** [comment]
The pages are live and working as intended (remember to add a trailing slack to the end of the URL though!). There may be 2 or 3 URL names that need to changed in the URL above.

1. Form 10-10ez-pdf (https://www.va.gov/find-forms/about-form-10-10ez-pdf/): I recall a slack convo to remove the "pdf" from the URL name. But doing so results in **2.**
2. Form 10-10ez (esp): https://www.va.gov/find-forms/about-form-10-10ez/ currently navigates to the Spanish version of this form. This URL is likely meant for 10-10ez-pdf.
3. Form 10-0137-espanol: The current url https://www.va.gov/find-forms/about-form-10-0137-espanol/ may need to be changed once a standardized URL naming convention for Spanish forms are decided on.

CC: @RLHecht

**Comment by jenniferlee-dsva** [comment]
1/ they're live - awesome! 
2/ thank you! 
3/ Just a reminder: the form page URLs - we can control; this is what we have standardized using @mnorthuis 's standard page URL pattern - `find-forms/about-[va-form-number]`.  The form PDF urls however - we cannot control; the form PDF urls are totally controled upstream by the form managers and is generated  by the form filename.

**Comment by jenniferlee-dsva** [comment]
FYI - @ncksllvn @kelsonic @lunascoop @RLHecht - The form detail page URLs are generated like this in the CMS -- see product status and decisions here: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/find-va-forms-redesign/form-detail-landing-pages/status-and-decisions.md#071520 

When we publish the forms, for this first launch, we manually unchecked the autogenerate URL box, so it is hard set. (After first launch, the checkbox logic in the CMS is supposed to automatically become unchecked after publish, so the URLs don't change even after/when the form file changes upstream in the source VA microsoft DB. 

So I think we have a CMS bug here. Attaching screenshot. The autogenerate URL checkbox is unchecked, but somehow, the URLs that's showing here IS different from what Randi and Selina hard set during review. 

![image](https://user-images.githubusercontent.com/43381063/92605185-6c4d6000-f27f-11ea-8c3a-e181aeb14278.png)

**Comment by jenniferlee-dsva** [comment]
The immediate solution for fixing these couple of form page URLs - I think - can be done in the CMS. Just by manually updating the URLs for these pages in the URL alias field. 

Long term - we need to pull in @swirtSJW on the CMS team to take a look at why or how the URL field changed even though we 
 had previously unchecked the auto generate checkbox.   <-- @lunascoop or @RLHecht : would you mind following up with Steve on the forms channel about that? (please @ me as well so I can follow).

**Comment by RLHecht** [comment]
@jenniferlee-dsva In looking through the revision log for these forms, it's possible that we never removed "pdf" from the URL, just in the H2 on the page. I'll still reach out to @swirtSJW to ensure there isn't a bug here. But in the meantime we'll manually fix the URLs so that they are showing correctly.

**Comment by lunascoop** [comment]
Perhaps I am missing something, but are there any other URLs that are incorrect/not showing correctly aside from 10-10ez-pdf? If this issue only pertains to form 10-10ez-pdf, is it possible that the URL change did not register/save since that URL (https://www.va.gov/find-forms/about-form-10-10ez/ ) was already assigned to the 10-10ez-esp form?

**Comment by lunascoop** [comment]
Following up on my previous comment, @RLHecht I noticed in ticket [#11336 ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/11336) you commented that you made a change to 10-10ez (/find-forms/about-form-10-10ez-pdf to /find-forms/about-form-10-10ez) on 7/31/20. I pasted your comment below. I can try to find the slack convo. Perhaps the issue is that when that change to the /find-forms/about-form-10-10ez-pdf URL name was made, Drupal didn't throw an error or pop-up that the URL was already in use. And, perhaps that was because the pages hadn't been published yet.

> This has been reviewed. I changed the one 10-10EZ that we discussed on slack. Everything else looked good, @jenniferlee-dsva FYSA @MickinSahni

**Comment by jenniferlee-dsva** [comment]
@lunascoop - I think I kinda sorta understand what may have occurred. It's a little hard to write out. 

I took a look at the legacy 10-10ez form detail page. 

- In the past, the legacy form detail page / https://www.va.gov/vaforms/form_detail.asp?formno=10ez /  included links to both the English 10-10EZ form + the EZR and Spanish forms. 
- If you go there today, it only shows the Spanish form. 

Because of the change on the legacy form page, I'm guessing that something happened at the source VA DB. When the form manager updated the form somehow, whatever change they made overrode something so that it set the Spanish form as the default or primary 10-10ez.  

I don't want to gumshoe what might have happened upstream, as we have very little control there!  

However, if you and Randi can try to manually fix this in the CMS, let's see if that resolves this issue at our end. And @RLHecht - yes, please do find out via Steve, if this is possibly a bug where unchecking the URL box doesn't actually prevent it from changing with future migrations. 

English - Form 10-10ez-pdf (Currently: https://www.va.gov/find-forms/about-form-10-10ez-pdf/) 

- This URL should change to `about-form-10-10ez`

Spanish - Form 10-10ez (esp): https://www.va.gov/find-forms/about-form-10-10ez/ currently navigates to the Spanish version of this form. This URL is likely meant for 10-10ez-pdf.

- This URL should change to `about-form-10-10ez-espanol ` or **whatever shema @mnorthuis recommends.**

Spanish Advance Directive - Form 10-0137-espanol: The current url https://www.va.gov/find-forms/about-form-10-0137-espanol/ may need to be changed once a standardized URL naming convention for Spanish forms are decided on.

- This URL seems okay, since this actually IS the Spanish version of the 10-0137. (Am I missing a problem here? LMK.)

**Comment by mnorthuis** [comment]
@jenniferlee-dsva  I'm inclined to say that adding `-espanol` to the end of the URL would be the recommended approach for form detail pages that have a spanish form on them.  I am a bit concerned that this may make it appear as though the page is in Spanish, but it looks like some of the descriptive content from the API is in spanish so maybe we are ok?

https://www.va.gov/find-forms/about-form-10-10ez-espanol/ 

Remind me, do we have control over the H1s?  If we do, we should establish consistent titles for Spanish forms as well.  I would recommend using "(en Español)".  It looks like some pages have "(esp)" or "(espanol)" or possibly other variations in the H1.

**Comment by RLHecht** [comment]
@mnorthuis We don't have control over the H1s. Just the form names. The H1s get pulled straight from the forms DB. I will update the above URL with the -espanol at the end.

**Comment by RLHecht** [comment]
These URLS have now been updated (the 10-10ez-espanol and 10-10ez (removing PDF from end). Should be live after today's deploy. FYSA @jenniferlee-dsva @lunascoop

**Comment by MickinSahni** [comment]
Closing this out as it's published. We have a separate ticket for resolving the 3 urls issue (#13396)

**Comment by jenniferlee-dsva** [comment]
Thank you - I see the live pages with the updated URLS. Looks good! @RLHecht @lunascoop @mnorthuis

**Comment by mnorthuis** [comment]
@jenniferlee-dsva 
Some questions in regards to the list of pages in the top description of this issue - to make sure these are accurate and that we get the corresponding redirects in #5549 correct:

21 & 39 - we should have 2 10-10ez pages `/find-forms/about-form-10-10ez/` and `/find-forms/about-form-10-10ez-espanol/`
  - we have 2 legacy URLs in our redirect list that need to be clarified 
  - www.va.gov/vaforms/form_detail.asp?formno=10ez  <-- this is an accurate URL that currently points at the spanish version, based on what Jen says above, it used to include both and I believe it should redirect to the english version
  - www.va.gov/vaforms/form_detail.asp?formno=10ez (pdf)  <-- this is an inacurrate URL and should be removed from our redirect list
  - I don't believe you can get to just the english form on the legacy site any longer, at least I couldn't find it, so the URL in the first bullet will be redirected to the new english detail page since its the primary URL.  There is no legacy spanish URL to redirect that I am aware of.  

33. /find-forms/about-form-21p-509
  - I do not see a redirect in our list for the legacy version of this page (https://www.va.gov/vaforms/form_detail.asp?formno=21p-509).  Should we add it?  

35 /find-forms/about-form-10-0137a
  - I do not see a redirect in our list for the legacy version of this page (https://www.va.gov/vaforms/form_detail.asp?formno=0137a).  Should we add it?  It is different than the 10-0137 and we do have a redirect listed for that. 
  - Also, the redirect list includes one legacy page slated to redirect to the english version of the new form - the legacy page currently shows both english and spanish forms. That seems accurate but want to be sure. 

37. /find-forms/about-form-21p-527
  - I do not see a redirect in our list for the legacy version of this page (https://www.va.gov/vaforms/form_detail.asp?formno=21p-527).  Should we add it?  It is different than the 527ez, that redirect is accounted for.

38. /find-forms/about-form-21p-534
  - I do not see a redirect in our list for the legacy version of this page (www.va.gov/vaforms/form_detail.asp?formno=21p-534).  Should we add it?  It is different than thte 534ez, that redirect is accounted for.

**Comment by mnorthuis** [comment]
Discussed the above with @jenniferlee-dsva and @brianalloyd 

**10-10EZ**
`www.va.gov/vaforms/form_detail.asp?formno=10ez` will redirect to /find-forms/about-form-10-10ez/

`www.va.gov/vaforms/form_detail.asp?formno=10ez (pdf)` is technically not a valid URL, the space and content in parentheses is ignored/dropped.  I will update this url in the redirect list to encode the space - `https://www.va.gov/vaforms/form_detail.asp?formno=10ez%20(pdf)` - and redirect this to the english version as well so any SEO value it has is captured.


`/find-forms/about-form-21p-509` will be handled in the bulk redirect to `/find-forms/`, no 1:1 redirect needed

`/find-forms/about-form-10-0137a` will be handled in the bulk redirect to `/find-forms/`, no 1:1 redirect needed

`/find-forms/about-form-21p-527` will be handled in the bulk redirect to `/find-forms/`, no 1:1 redirect needed

`/find-forms/about-form-21p-534` will be handled in the bulk redirect to `/find-forms/`, no 1:1 redirect needed
